regression_detector.py

```diff
@@ -158,6 +159,31 @@ def process_regressions_into_gh_issue(regressions_dict, owner: str, output_path:
         f.write(issue_body)
 
 
+def get_best_start_date(latest_metrics_jsons, end_date: str) -> Optional[datetime]:
+    """Get the date closest to `end_date` from `latest_metrics_jsons`"""
+    end_datetime = datetime.strptime("%Y-%m-%d", end_date)
+    for metrics_json in latest_metrics_jsons:
+        start_datetime = datetime.strptime(metrics_json.split('/')[-1].split('-')[-1].split('.')[0], '%Y%m%d%H%M%S')
+        if start_datetime < end_datetime:
+            return start_datetime
+    return None
+
+
+def get_metrics_by_date(latest_metrics_jsons: List[str], pick_date: datetime):
+    pick_metrics_json_key: Optional[str] = None
+    for metrics_json in latest_metrics_jsons:
+        metric_datetime = datetime.strptime(metrics_json.split('/')[-1].split('-')[-1].split('.')[0],
+                                                            '%Y%m%d%H%M%S')
+        # Use the latest metric file on on the same day
```

Comment:
In this case, would it be more effective to reverse the latest_metrics_jsons (or traverse it in reverse order) and just return when the first hit occurs?

^ definitely a nit

Reply:
Sorry I don't get this, I believe `latest_metrics_jsons ` is already sorted in the reverse order?

Reply:
Oh, in this case, shouldn't you break after a value is found? Otherwise, if multiple jsons are found on the same day, won't the earlier ones overwrite the key?

Reply:
Ah, I see what you mean. Yes, a loop break is needed here. Thanks for pointing it out!